spock-core/src/main/java/org/spockframework/mock/constraint/NamedArgumentListConstraint.java

```diff
@@ -51,7 +54,7 @@ public boolean isSatisfiedBy(IMockInvocation invocation) {
   @SuppressWarnings("unchecked")
   public String describeMismatch(IMockInvocation invocation) {
     List<Object> args = invocation.getArguments();
-    if (args.size() == 1 && args.get(0) instanceof Map)
+    if ((args.size() == 1 || isMixed) && args.get(0) instanceof Map)
```

Comment:
I think it would make sense to extract this condition into a method and use it in both places.